src/plugins/bsodmon/bugcheck.cpp

```diff
@@ -0,0 +1,488 @@
+*                                                                         *
+* If you have any questions about the licensing restrictions on using     *
+* DRAKVUF in other works, are happy to help.  As mentioned above,         *
+* alternative license can be requested from the author to integrate       *
+* DRAKVUF into proprietary applications and appliances.  Please email     *
+* tamas.k.lengyel@gmail.com for further information.                      *
+*                                                                         *
+* If you have received a written license agreement or contract for        *
+* Covered Software stating terms other than these, you may choose to use  *
+* and redistribute Covered Software under those terms instead of these.   *
+*                                                                         *
+* Source is provided to this software because we believe users have a     *
+* right to know exactly what a program is going to do before they run it. *
+* This also allows you to audit the software for security holes.          *
+*                                                                         *
+* Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+* and add new features.  You are highly encouraged to submit your changes *
+* on https://github.com/tklengyel/drakvuf, or by other methods.           *
+* By sending these changes, it is understood (unless you specify          *
+* otherwise) that you are offering unlimited, non-exclusive right to      *
+* reuse, modify, and relicense the code.  DRAKVUF will always be          *
+* available Open Source, but this is important because the inability to   *
+* relicense code has caused devastating problems for other Free Software  *
+* projects (such as KDE and NASM).                                        *
+* To specify special license conditions of your contributions, just say   *
+* so when you send them.                                                  *
+*                                                                         *
+* This program is distributed in the hope that it will be useful, but     *
+* WITHOUT ANY WARRANTY; without even the implied warranty of              *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+* license file for more details (it's in a COPYING file included with     *
+* DRAKVUF, and also available from                                        *
+* https://github.com/tklengyel/drakvuf/COPYING)                           *
+*                                                                         *
+***************************************************************************/
+
+#include <libdrakvuf/libdrakvuf.h>
+
+#include "bsodmon.h"
+
+/*
+ * Bugcheck code list:
+ *     https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-check-code-reference2
+ */
+
+void init_bugcheck_win7( bsodmon* monitor )
```

Comment:
Instead of doing this during initialization, just convert this to a `static const char*` array similar to the ones you find in the syscall plugin.

Reply:
I'm afraid this case is different than the syscalls one, there are several gaps, e.g.:
```
0x000000A5 | ACPI_BIOS_ERROR
0x000000A7 | BAD_EXHANDLE
0x000000AB | SESSION_HAS_VALID_POOL_ON_EXIT
[...]
0x000000AD | VIDEO_DRIVER_DEBUG_REPORT_REQUEST
0x000000B1 | BGI_DETECTED_VIOLATION
0x000000B4 | VIDEO_DRIVER_INIT_FAILURE
0x000000B8 | ATTEMPTED_SWITCH_FROM_DPC
[...]
0x00000106 | AGP_ILLEGALLY_REPROGRAMMED
0x00000108 | THIRD_PARTY_FILE_SYSTEM_FAILURE
0x00000109 | CRITICAL_STRUCTURE_CORRUPTION
[...]
0x0000013D | CRITICAL_INITIALIZATION_FAILURE
0x00000140 | STORAGE_DEVICE_ABNORMALITY_DETECTED
[...]
0x000001D4 | UCMUCSI_LIVEDUMP
0x00000356 | XBOX_ERACTRL_CS_TIMEOUT
0x00000BFE | BC_BLUETOOTH_VERIFIER_FAULT
0x00000BFF | BC_BTHMINI_VERIFIER_FAULT
0x00020001 | HYPERVISOR_ERROR
0x1000007E | SYSTEM_THREAD_EXCEPTION_NOT_HANDLED_M
```

Reply:
I see. Alright.